src/instructlab/cli/model/upload.py

```diff
@@ -0,0 +1,249 @@
+        upload_dest: str,
+        release: str,
+        hf_token: str,
+    ) -> None:
+        super().__init__(
+            ctx=ctx,
+            model=model,
+            model_dir=model_dir,
+            upload_dest=upload_dest,
+            release=release,
+        )
+        self.hf_token = (
+            hf_token.strip()
+        )  # Remove trailing whitespaces from the token if they exist
+        self.local_model_path = ""
+
+    def upload(self):
+        """
+        Upload specified model to Hugging Face
+        """
+
+        # HF token check
+        if len(self.hf_token) == 0:
+            raise ValueError(
+                """HF_TOKEN var needs to be set in your environment to upload HF Model.
+                Alternatively, the token can be passed with --hf-token flag.
+                The HF Token is used to authenticate your identity to the Hugging Face Hub."""
+            )
+
+        click.echo(
+            f"Uploading model to Hugging Face:\n{DEFAULT_INDENT}Model: {self.model}\n{DEFAULT_INDENT}Destination: {self.upload_dest}@{self.release}\n"
+        )
+
+        # set logger
+        if self.ctx.obj is not None:
+            hf_logging.set_verbosity(self.ctx.obj.config.general.log_level.upper())
+
+        # determine if model is path or name
+        if os.path.exists(self.model):
+            self.local_model_path = Path(self.model)
+            self.model = self.model.split("/")[-1]
+        else:
+            self.local_model_path = os.path.join(self.model_dir, self.model)
+            # throw an error if the built path is invalid
+            if not os.path.exists(self.local_model_path):
+                click.secho(
+                    f"Couldn't find model at {self.local_model_path} - are you sure it exists?",
+                    fg="red",
+                )
+                raise click.exceptions.Exit(1)
+
+        # upload either safetensors or gguf, if valid
+        if is_model_safetensors(self.local_model_path):
+            self.upload_safetensors()
+        elif is_model_gguf(self.local_model_path):
+            self.upload_gguf()
+        else:
+            click.secho(
+                f"Local model path {self.local_model_path} is a valid path, but is neither safetensors nor a GGUF - cannot upload to Hugging Face",
+                fg="red",
+            )
+            raise click.exceptions.Exit(1)
+
+    def upload_gguf(self) -> None:
+        try:
+            hf_api = HfApi()
+            resp = hf_api.upload_file(
+                path_or_fileobj=self.local_model_path,
+                path_in_repo=self.model,
+                repo_id=self.upload_dest,
+                revision=self.release,
+                token=self.hf_token,
+            )
+            logger.debug(f"Hugging Face response:\n{resp}")
+        except Exception as exc:
+            click.secho(
+                f"\nUploading GGUF model failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
```

Comment:
Might be handy to spit out the model name or path as part of this debug message. E.g:

```suggestion
                f"\nUploading GGUF model {self.local_model_path} failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
```

Reply:
Hmmm, originally I didn't include this because it was included in the original output, but looking now that doesn't actually include `local_model_path` as the var hasn't been built out yet - so makes sense to me!